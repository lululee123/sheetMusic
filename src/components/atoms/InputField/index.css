.input-field {
	width: 100%;

	& .top {
		display: flex;
		justify-content: space-between;
		margin-bottom: 4px;

		& .title {
			font-size: 14px;
			font-weight: 500;
			line-height: 1.71;
			color: var('--darkBlack');
		}

		& .must-fill {
			font-size: 14px;
			line-height: 1.71;
			text-align: right;
			color: var('--gray');
		}
	}

	& .input {
		width: 100%;
		height: 40px;
		border-radius: 2px;
		border: solid 1px var('--gray');
		box-sizing: border-box;
		padding: 8px 12px;
		font-size: 14px;
		background-color: transparent;
	}

	& .error-input {
		border-color: var('--red');
	}

	& .error-box {
		display: flex;
		margin-top: 4px;

		& .error-text {
			font-size: 14px;
			font-weight: 500;
			line-height: 1.71;
			color: var('--red');
			margin-left: 8px;
		}
	}
}
