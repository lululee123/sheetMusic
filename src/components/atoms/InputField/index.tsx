import React, { ChangeEvent } from 'react';
import classnames from 'classnames';
import { useTranslation } from 'react-i18next';

import Error from 'images/icon/error.inline.svg';

import styles from './index.css';

interface InputProperty {
	className?: string;
	title?: string;
	placeholder?: string;
	type?: string;
	value?: string;
	onChange?: (event: ChangeEvent<HTMLInputElement>) => void;
	mustFill?: boolean;
	error?: boolean;
	errorText?: string;
}

const InputField: React.FC<InputProperty> = ({
	className,
	title = '',
	placeholder = '',
	type = 'text',
	value = '',
	onChange,
	mustFill = false,
	error = false,
	errorText = '',
}) => {
	const { t } = useTranslation(['auth']);
	return (
		<div className={classnames(styles.inputField, className)}>
			{(title || mustFill) && (
				<div className={styles.top}>
					<span className={styles.title}>{title}</span>
					{mustFill && <span className={styles.mustFill}>{t('mustFill')}</span>}
				</div>
			)}

			<input
				placeholder={placeholder}
				type={type}
				className={classnames(styles.input, error && styles.errorInput)}
				value={value}
				onChange={onChange}
			/>
			{error && (
				<div className={styles.errorBox}>
					<Error />
					<div className={styles.errorText}>{errorText}</div>
				</div>
			)}
		</div>
	);
};

export default InputField;
