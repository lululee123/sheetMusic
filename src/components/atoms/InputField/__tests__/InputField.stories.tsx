import React from 'react';
import { storiesOf } from '@storybook/react';
import { text, boolean } from '@storybook/addon-knobs';
import { action } from '@storybook/addon-actions';

import InputField from 'components/atoms/InputField';

const stories = storiesOf('atoms/InputField', module);

stories.add('__interactive', () => (
	<div style={{ width: '300px' }}>
		<InputField
			title={text('title', '標題')}
			placeholder={text('placeholder', 'placeholder')}
			type={text('type', 'text')}
			value={text('value', 'test')}
			onChange={action('onChange')}
			mustFill={boolean('mustFill', true)}
		/>
	</div>
));

stories.add('with error', () => (
	<div style={{ width: '300px' }}>
		<InputField
			title={text('title', '標題')}
			placeholder={text('placeholder', 'placeholder')}
			type={text('type', 'text')}
			value={text('value', 'test')}
			onChange={action('onChange')}
			mustFill={boolean('mustFill', true)}
			error={boolean('error', true)}
			errorText={text('errorText', '錯誤')}
		/>
	</div>
));

stories.add('with type password', () => (
	<div style={{ width: '300px' }}>
		<InputField
			title={text('title', '標題')}
			placeholder={text('placeholder', 'placeholder')}
			type={text('type', 'password')}
			value={text('value', 'test')}
			onChange={action('onChange')}
			mustFill={boolean('mustFill', true)}
		/>
	</div>
));

stories.add('without must fill', () => (
	<div style={{ width: '300px' }}>
		<InputField
			title={text('title', '標題')}
			placeholder={text('placeholder', 'placeholder')}
			type={text('type', 'password')}
			value={text('value', 'test')}
			onChange={action('onChange')}
			mustFill={boolean('mustFill', false)}
		/>
	</div>
));
