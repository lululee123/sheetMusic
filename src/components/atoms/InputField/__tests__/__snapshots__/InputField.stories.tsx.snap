// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`__interactive 1`] = `
<DocumentFragment>
  <div
    style="width: 300px;"
  >
    <div
      class="inputField"
    >
      <div
        class="top"
      >
        <span
          class="title"
        >
          標題
        </span>
        <span
          class="mustFill"
        >
          mustFill
        </span>
      </div>
      <input
        class="input"
        placeholder="placeholder"
        type="text"
        value="test"
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`with error 1`] = `
<DocumentFragment>
  <div
    style="width: 300px;"
  >
    <div
      class="inputField"
    >
      <div
        class="top"
      >
        <span
          class="title"
        >
          標題
        </span>
        <span
          class="mustFill"
        >
          mustFill
        </span>
      </div>
      <input
        class="input errorInput"
        placeholder="placeholder"
        type="text"
        value="test"
      />
      <div
        class="errorBox"
      >
        <test-file-stub />
        <div
          class="errorText"
        >
          錯誤
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`with type password 1`] = `
<DocumentFragment>
  <div
    style="width: 300px;"
  >
    <div
      class="inputField"
    >
      <div
        class="top"
      >
        <span
          class="title"
        >
          標題
        </span>
        <span
          class="mustFill"
        >
          mustFill
        </span>
      </div>
      <input
        class="input"
        placeholder="placeholder"
        type="password"
        value="test"
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`without must fill 1`] = `
<DocumentFragment>
  <div
    style="width: 300px;"
  >
    <div
      class="inputField"
    >
      <div
        class="top"
      >
        <span
          class="title"
        >
          標題
        </span>
      </div>
      <input
        class="input"
        placeholder="placeholder"
        type="password"
        value="test"
      />
    </div>
  </div>
</DocumentFragment>
`;
